README.md

```diff
@@ -100,6 +103,112 @@ overview of all settings configurable through the CLI.
 | `S3.KeypairsV4 (DEPRECATED)`                      | V4 keypairs for S3                                   | -                                 | -                                  | -            | `s3.keypairsV4`                     |
 
+### Single-Node Setup
+
+A single-node setup involves running all components (bus, worker, and autopilot)
+on the same machine. This is ideal for testing, development, or small-scale
+deployments. This setup is the default when running `renterd` without any flags.
+
+### Cluster Setup
+
+In a cluster setup, the bus, worker, and autopilot run on separate nodes. This
+setup is ideal for large-scale deployments where you want to horizontally scale
+your renter. The worker nodes can be spread across multiple machines, and the
+autopilot can be run on a separate machine.
+
+#### Bus Node Configuration
+
+The bus is the only node that exposes the UI. To run the bus separately, the
+autopilot and worker have to be disabled using the `--autopilot.enabled` and
+`--worker.enabled` flags. The only other requirement to run a bus is the (walet)
+seed.
+
+#### Worker Node Configuration
+
+To configure the worker as a standalone node, the autopilot has to be disabled
+using the `--autopilot.enabled` flag, and the bus has to be disabled. There's no
+flag to explicitly disable the `bus`, it's implied by configuring a remote
+address for the bus using the `--bus.remoteAddr` and `--bus.remotePassword`
+flags. When the bus is remote, the worker has to be configured with an external
+address of the form http://<worker-ip>:<port>, on localhost however this can be
+the same as the worker's HTTP address. The worker needs to know its location on
+the network because it relies on some webhooks it needs to register with the
+bus, which in turn needs to know how to reach the worker when certain events
+occur. Therefor it is important to start the worker after the bus is reachable.
+
+#### Autopilot Node Configuration
+
+To run the autopilot separately, the worker has to be disabled using the
+`--worker.enabled` flag. Similar to the worker, the autopilot has to be
+configured with a remote bus for the node not to start a bus itself. Alongside
+with knowing where the bus is located, the autopilot also has to be aware of the
+workers. These remote workers can be configured through yaml under the option
+`worker.remotes`, or through environment variables
+(`RENTERD_WORKER_REMOTE_ADDRS` and `RENTERD_WORKER_API_PASSWORD`).
+
+#### Example docker-compose with minimal configuration
+
+```yaml
+version: '3.9'
+
+services:
+  bus:
+    image: ghcr.io/siafoundation/renterd:master
+    container_name: renterd_bus
+    environment:
+      - RENTERD_SEED=<enter seed here>
+      - RENTERD_API_PASSWORD=bus-pass
+    ports:
+      - "9980:9980"
+      - "9981:9981"
+
+  worker-1:
```

Comment:
workers are missing the S3 port it seems

Reply:
exported the default S3 port, was trying to keep the docker-compose file to an absolute minimum but it does make sense to export those